pkg/kv/kvserver/gc/gc.go

```diff
@@ -706,6 +732,114 @@ func processAbortSpan(
 	})
 }
 
+type rangeBatcher struct {
+	gcer      GCer
+	now       hlc.Timestamp
+	threshold hlc.Timestamp
+	batchSize int64
+
+	pending     []storage.MVCCRangeKey
+	pendingSize int64
+}
+
+func (b *rangeBatcher) addAndMaybeFlushRangeFragment(
+	ctx context.Context, unsafeRange storage.MVCCRangeKey,
+) error {
+	rangeSize := int64(len(unsafeRange.StartKey)) + int64(len(unsafeRange.EndKey)) + storage.MVCCVersionTimestampSize
+	if len(b.pending) > 0 && (b.pendingSize+rangeSize) >= b.batchSize {
+		if err := b.flushPendingFragments(ctx); err != nil {
+			return err
+		}
+	}
+	// Count all fragments as we want to reduce cleanup batch size in raft
+	// downstream from GCer which would have to remove fragments.
```

Comment:
Good that we take this into account. However, we're not considering all of the historical versions. If we have e.g. `[a-c)` then that'll count for about 10 bytes here, but it could in principle have a million versions, which would result in a Raft batch of 10 MB.

I see that we have the same problem with point keys. I think we only get away with it because we use a miniscule batch size of 256 KB. That's probably fine for now, but we should add a `TODO` to fix this for both key types at some point, so it doesn't suddently blow up when someone GCs a long key history (e.g. by dropping a long GC TTL to a short one for an update-heavy table).